src/xhr.ts

```diff
@@ -41,6 +41,52 @@ export class JQueryRequestor extends Requestor {
   }
 }
 
+
+/**
+ * Uses fetch API to make Ajax requests
+ */
+export class FetchRequestor extends Requestor {
+  xhr<T>(settings: JQueryAjaxSettings): Promise<T> {
+    let url: URL = new URL(<string>settings.url);
+    let requestInit: RequestInit = {};
+    requestInit.method = settings.method;
+    requestInit.mode = 'cors';
+
+    if (settings.data) {
+      if (settings.method && settings.method.toUpperCase() === 'POST') {
+        requestInit.body = <string>settings.data;
+      } else {
+        let searchParams = new URLSearchParams(settings.data);
+        searchParams.forEach((value, key) => {
+          url.searchParams.append(key, value);
+        });
+      }
+    }
+
+    // Calculate 'Accept' header value
+    let headers = settings.headers || {};
+    if (settings.dataType && settings.dataType.toLowerCase() === 'json') {
+      headers['Accept'] = 'application/json, text/javascript, */*; q=0.01';
+    } else {
+      headers['Accept'] = '*/*';
+    }
+
+    // Set the request headers
+    requestInit.headers = {};
+    for (let i in headers) {
+      requestInit.headers[i] = <string>headers[i];
+    }
+
+    return fetch(url.toString(), requestInit).then(response => {
+      if (settings.dataType && settings.dataType.toLowerCase() === 'json') {
```

Comment:
I also would check for the response type from the server:

```ts
const contentType = response.headers.get('content-type');
if ((settings.dataType && settings.dataType.toLowerCase() === 'json') || (contentType !== null && contentType.includes('application/json'))) {
```

Reply:
done!